### PR TITLE
More specific pseudotag replace

### PR DESCRIPTION
**Pull request by tenbradley**
Kind of a hack.
Matching "<"+pseudotag"+" " or "</"+pseudotag"+" " for replacing pseudotags.

### CODE REVIEW

File: framework/HypermediaResult.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		}
+		strings.pop();
+		if (strings.length !== 0) {
+			mainStrings.push(`</${tag}>`);
+			mainValues.push(html(strings, values));
+		} else {
+			mainStrings[mainStrings.length - 1] += `</${tag}>`;
+		}
+
+		//dispose(state);
```
Comment (m6jones) [inline framework/HypermediaResult.js:142]: You can remove this

**Review by m6jones — APPROVED** [review]

File: framework/HypermediaResult.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+export class HypermediaResult extends TemplateResult {
+
+	constructor(strings, values, type, processor) {
+		super(strings, values, type, processor);
+		this.processing();
+	}
+
+	/*
+	* Function called on HypermediaResult object to get the associated html
+	* Should process the psuedocode if this is the first call to getHTML
```
Comment (zinabat) [inline framework/HypermediaResult.js:28]:
```suggestion
	* Should process the base tag if this is the first call to getHTML
```
I think you meant this?

File: framework/HypermediaResult.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	* otherwise, retrieves previous processing
+	*/
+	getHTML() {
+		return super.getHTML();
+	}
+
+	processing() {
```
Comment (zinabat) [inline framework/HypermediaResult.js:35]: I would still love to see more comments throughout this function

File: framework/HypermediaResult.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	render(state, components, resources, pseudoTag, strings, values) {
+		const tag = components.componentTag(resources.classes);
+		if (!tag) return null;
+		strings = [...strings];
+		const mainStrings = [], mainValues = [];
+		while (strings.length > 0) {
+			let string = strings.shift();
+			if (!string) break;
+			string = string.replace(`<${pseudoTag} `, `<${tag} `);
+			string = string.replace(`</${pseudoTag} `, `</${tag} `);
```
Comment (zinabat) [inline framework/HypermediaResult.js:120]: I assume this fixes #17

File: framework/hypermedia-components.js [hunk 5]
```diff
@@ -1,121 +1,10 @@
...
 
 export function customHypermediaElement(tag, elementClass, pseudoTag, hypermediaClasses, options) {
-	pseudoTag = pseudoTag || tag;
-	const components = componentStoreFactory(pseudoTag);
-	components.register(tag, hypermediaClasses);
-	customElements.define(tag, elementClass, options);
+	custom(tag, elementClass, pseudoTag, hypermediaClasses, options);
 }
 
 export function html(strings, ...values) {
```
Comment (zinabat) [inline framework/hypermedia-components.js:8]: It's so much clearer what's happening here. 🎉

File: framework/HypermediaResult.js (hunk 4 above)
Comment (tenbradley) [reply framework/HypermediaResult.js:120]: That is the intent. Not sure if there is a better way.

File: framework/HypermediaResult.js (hunk 3 above)
Comment (tenbradley) [reply framework/HypermediaResult.js:35]: I agree. I don't think this function will change much in trying to improve everything, so it is likely a good idea to comment it now.
I can work on this.

**Pull request merged**